logic/account.js

```diff
@@ -356,7 +356,7 @@ class Account {
 
 				if (performComputationFor.includes('approval')) {
 					rows.forEach(accountRow => {
-						accountRow.approval = self.calculateApproval(
+						accountRow.approval = getAllSelf.calculateApproval(
```

Comment:
You're right but this it's completely out of the scope and it will produce massive changes in this file. I prefer to stick to the purpose of the issue.